lib/utils.ts

```diff
@@ -26,15 +26,15 @@ function isMergeableObject(value: unknown): boolean {
  * @param shouldRemoveNullObjectValues - If true, null object values will be removed.
  * @returns - The merged object.
  */
-function mergeObject<TTarget extends unknown[] | Record<string, unknown>>(target: TTarget, source: TTarget, shouldRemoveNullObjectValues = true): TTarget {
+function mergeObject<TValue extends OnyxValue<OnyxKey>>(target: OnyxEntry<TValue>, source: TValue, shouldRemoveNullObjectValues = true): TValue {
```

Comment:
Should we restrict this function to only onyx values? I think it was supposed to be used as a generic utils